### PR TITLE
Print a ready line, fix Python 2.7 issue, remove some console logs.

### PR DESCRIPTION
**Pull request by scottwittenburg**
Also includes an attempt to fix the issue where no explicit return
from an rpc method results in the rpc method never returning, but
this may not completely solve the problem.

### CODE REVIEW

**Comment by scottwittenburg** [comment]
@aron-helser I think my attempt to avoid the need for a return value broke some tests, but this may not have been what we wanted anyway.

File: python/src/wslink/websocket.py [hunk 1]
```diff
@@ -172,6 +172,12 @@ def setServerProtocol(self, newServerProtocol):
     def getServerProtocol(self):
         return self._protocolHandler
 
+    ### Does not seem to work to print a "ready line"
+    # def startFactory(self):
```
Comment (aronhelser) [inline python/src/wslink/websocket.py]: Debugging, Twisted seems to call startFactory on a twisted.web.server.Site object, so http.py HttpServer::startFactory() gets called. We might be able to provide an override for that in wslink.

File: python/src/wslink/server.py [hunk 2]
```diff
@@ -220,6 +224,8 @@ def start_webserver(options, protocol=wsl.ServerProtocol, disableLogging=False):
     # Initialize testing: checks if we're doing a test and sets it up
     # testing.initialize(options, reactor, stop_webserver)
 
+    reactor.callWhenRunning(print_ready)
```
Comment (aronhelser) [inline python/src/wslink/server.py]: I'm concerned this isn't the same as the factory being ready.

File: python/src/wslink/websocket.py [hunk 3]
```diff
@@ -285,7 +290,7 @@ def sendWrappedMessage(self, rpcid, content, method=''):
         wrapper = {
             "wslink": "1.0",
             "id": rpcid,
-            "result": content,
+            "result": content if content else [],
```
Comment (aronhelser) [inline python/src/wslink/websocket.py]: The test failures are just looking for specific return values. But I'd like to understand why a return value is required.

Comment (aronhelser) [inline python/src/wslink/websocket.py]: OK, I think the error here is in session.js: onMessage
if (payload.result) will fail for a 'falsy' result, and it won't resolve the promise. I'll fix that.

File: python/src/wslink/server.py (hunk 2 above)
Comment (scottwittenburg) [inline python/src/wslink/server.py]: You may know more about it than I do, but it seems to me that once the reactor is running, we should be good to go:

https://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IReactorCore.html#callWhenRunning

We used the hook above to know whether our test thread should try to connect to the running server, and that seemed to be reliable.

**Comment by scottwittenburg** [comment]
I don't see anything in here that stands out as a possible issue.

+1

**Pull request merged**